### PR TITLE
Issue#48 improve test coverage and test infrastructure

### PR DESCRIPTION
**Pull request by Waginator**
33 Commits and 73 Files changed, have fun with it :P

### CODE REVIEW

File: src/test/java/com/metalr2/model/user/UserEntityTest.java [hunk 1]
```diff
@@ -1,143 +1,183 @@
...
+      assertThat(setEmptyCollection).hasMessage("At least one user role must be set!");
+    }
 
-  @Test
-  void updateOfEmailShouldBePossible() {
-    String initialEmail = "test@test.com";
-    String newEmail     = "test-update@test.com";
+    @Test
+    @DisplayName("updating the user roles with an null set should throw an exception")
+    void update_of_user_roles_with_null_value_should_throw_exception() {
```
Comment (hedgehog1833) [inline src/test/java/com/metalr2/model/user/UserEntityTest.java]: This method also tests for an empty set instead of null.

File: src/test/java/com/metalr2/model/BaseEntityIT.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    assertThat(testEntity.getCreatedBy()).isNull();
+    assertThat(testEntity.getCreatedDateTime()).isNull();
+    assertThat(testEntity.getLastModifiedBy()).isNull();
+    assertThat(testEntity.getLastModifiedDateTime()).isNull();
+
+    testRepository.save(testEntity);
+
+    assertThat(testEntity.getCreatedBy()).isEqualTo(AUDITOR_USER);
+    assertThat(testEntity.getCreatedDateTime()).isCloseTo(LocalDateTime.now(), offset);
```
Comment (hedgehog1833) [inline src/test/java/com/metalr2/model/BaseEntityIT.java]: Seems nicer to me to put LocalDateTime.now() into a local variable. It is used twice and it compares the exact same moment in time as the createdDateTime and lastModifiedDateTime should be equal.

File: src/test/java/com/metalr2/MetalReleaseRadarApplicationIT.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.metalr2;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+
+@SpringBootTest
+@TestPropertySource(locations = "classpath:application-integrationtest.properties")
+class MetalReleaseRadarApplicationIT {
```
Comment (hedgehog1833) [inline src/test/java/com/metalr2/MetalReleaseRadarApplicationIT.java]: Integrationtest tag annotation missing?

File: pom.xml [hunk 4]
```diff
@@ -46,6 +46,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <jsonwebtoken.version>0.9.1</jsonwebtoken.version>
         <junit-jupiter.version>5.5.0-M1</junit-jupiter.version>
+        <mockito.version>2.23.0</mockito.version>
```
Comment (hedgehog1833) [inline pom.xml]: Typo in line 53. For some reason the github review-thingy does not highlight this line as new here.
Or I am stupid.

File: src/test/java/com/metalr2/service/email/JavaMailEmailServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    when(templateEngine.process(anyString(), any(Context.class))).thenReturn("<h1>Test</h1>");
+  }
+
+  @AfterEach
+  void tearDown() {
+    reset(emailSender, templateEngine, mailConfig);
+  }
+
+  @Test
+  @DisplayName("")
```
Comment (hedgehog1833) [inline src/test/java/com/metalr2/service/email/JavaMailEmailServiceTest.java]: Display name for send_email is empty

File: src/test/java/com/metalr2/web/controller/discogs/ArtistSearchRestClientTest.java [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @DisplayName("Tests for method searchByName()")
+  class SearchByNameTest {
+
+    @Test
+    @DisplayName("Searching by name should return a valid result")
+    void search_by_name() {
+      // given
+      final String ARTIST_NAME_QUERY = "Nirvana";
```
Comment (hedgehog1833) [inline src/test/java/com/metalr2/web/controller/discogs/ArtistSearchRestClientTest.java]: Nirvana? I think we should put something like Darkthrone in here. We are not here for fun :)

File: src/test/java/com/metalr2/model/user/UserEntityTest.java (hunk 1 above)
Comment (Waginator) [reply src/test/java/com/metalr2/model/user/UserEntityTest.java]: The evil of Copy & Paste

File: src/test/java/com/metalr2/model/BaseEntityIT.java (hunk 2 above)
Comment (Waginator) [reply src/test/java/com/metalr2/model/BaseEntityIT.java]: fixed

File: src/test/java/com/metalr2/service/email/JavaMailEmailServiceTest.java (hunk 5 above)
Comment (Waginator) [reply src/test/java/com/metalr2/service/email/JavaMailEmailServiceTest.java]: Fixed. I also removed the MimeMessageArgumentCaptor. It was useless.

**Comment by hedgehog1833** [comment]
Closes #48
